src/test/java/com/wikia/webdriver/testcases/desktop/anonymization/AnonymizationTests.java

```diff
@@ -0,0 +1,194 @@
+
+    articleHistoryPage.logOut();
+
+    SpecialAnonymizationUserPage anonymizationPage = new SpecialAnonymizationUserPage().open();
+    anonymizationPage.loginAs(User.SUS_STAFF);
+    anonymizationPage.fillFutureAnon(qanon).submitAnonymization();
+
+    Assertion.assertStringContains(anonymizationPage.getAnonConfirmation(), qanon);
+
+    articleHistoryPage = new ArticleHistoryPage().open(testsite);
+    Assertion.assertFalse(articleHistoryPage.isUserInHistory(user.getUsername()));
+  }
+
+  @Test
+  public void anonymizationMessageWallRemoved() {
+    Credentials credentials = new Credentials();
+    String timestamp = Long.toString(DateTime.now().getMillis());
+    String testsite = "AnonymizationTest";
+    String qanon = User.SUS_REGULAR_USER.getUserName() + timestamp;
+    String passw = User.SUS_REGULAR_USER.getPassword();
+    SignUpUser user = new SignUpUser(qanon,
+                                     credentials.emailAnonymousUserTestWikia,
+                                     passw,
+                                     LocalDate.of(1990, 3, 19)
+    );
+    UserRegistration.registerUserEmailConfirmed(user);
+
+    new WikiBasePageObject().loginAs(user.getUsername(), wikiURL);
+
+    MessageWall wall = new MessageWall().open(user.getUsername());
+    MiniEditorComponentObject mini = wall.triggerMessageArea(true);
+    String message = String.format("%s%s", PageContent.MESSAGE_WALL_MESSAGE_PREFIX, timestamp);
+    String title = String.format("%s%s", PageContent.MESSAGE_WALL_TITLE_PREFIX, timestamp);
+    mini.switchAndWrite(message);
+    wall.setTitle(title);
+    wall.submit();
+    wall.verifyMessageText(title, message, user.getUsername());
+    wall.triggerEditMessageArea();
+    String messageEdit = PageContent.MESSAGE_WALL_MESSAGE_EDIT_PREFIX + Long.toString(DateTime.now()
+                                                                                          .getMillis());
+    mini.switchAndEditMessageWall(messageEdit);
+    wall.submitEdition();
+    wall.verifyMessageEditText(title, messageEdit, user.getUsername());
+
+    try {
+      Thread.sleep(15000);
+    } catch (InterruptedException e) {
+      Log.logError("Interruption during waiting for Message Wall background task", e);
+    }
+
+    new ArticleContent(user.getUsername()).push("Test" + timestamp, testsite);
+
+    ArticleHistoryPage articleHistoryPage = new ArticleHistoryPage().open(testsite);
+    String id = articleHistoryPage.getHistoryID(user.getUsername());
+    Assertion.assertTrue(articleHistoryPage.isUserInHistory(user.getUsername()));
+
+    articleHistoryPage.logOut();
+
+    SpecialAnonymizationUserPage anonymizationPage = new SpecialAnonymizationUserPage().open();
+    anonymizationPage.loginAs(User.SUS_STAFF);
+    anonymizationPage.fillFutureAnon(qanon).submitAnonymization();
+
+    Assertion.assertStringContains(anonymizationPage.getAnonConfirmation(), qanon);
+
+    try {
+      Thread.sleep(5000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.sjc-dev.k8s.wikia.net/coding_rules#rule_key=squid%3AS2925)

[project-id]: com.wikia.selenium-tests